File: src/main/java/org/model/Competences/HistoriqueCompetence.java
package org.model.Competences;

import javafx.beans.property.BooleanProperty;
import javafx.beans.property.SimpleBooleanProperty;

import java.time.LocalDateTime;

public class HistoriqueCompetence {

    private int id;
    private int competenceId;

    private String category;
    private String type;
    private String description;

    private String niveau;
    private int anneesExperience;
    private String certification;
    private String statut;

    private String email;

    private String statusChange;
    private LocalDateTime dateModification;

    // ✅ Checkbox selection (JavaFX)
    private final BooleanProperty selected = new SimpleBooleanProperty(false);

    public HistoriqueCompetence() {}

    public HistoriqueCompetence(int id, int competenceId,
                                String category, String type, String description,
                                String niveau, int anneesExperience,
                                String certification, String statut,
                                String email,
                                String statusChange,
                                LocalDateTime dateModification) {

        this.id = id;
        this.competenceId = competenceId;
        this.category = category;
        this.type = type;
        this.description = description;
        this.niveau = niveau;
        this.anneesExperience = anneesExperience;
        this.certification = certification;
        this.statut = statut;
        this.email = email;
        this.statusChange = statusChange;
        this.dateModification = dateModification;
    }

    /* =========================
       GETTERS
       ========================= */

    public int getId() { return id; }
    public int getCompetenceId() { return competenceId; }

    public String getCategory() { return category; }
    public String getType() { return type; }
    public String getDescription() { return description; }

    public String getNiveau() { return niveau; }
    public int getAnneesExperience() { return anneesExperience; }
    public String getCertification() { return certification; }
    public String getStatut() { return statut; }

    public String getEmail() { return email; }
    public String getStatusChange() { return statusChange; }
    public LocalDateTime getDateModification() { return dateModification; }

    /* =========================
       SETTERS
       ========================= */

    public void setId(int id) { this.id = id; }
    public void setCompetenceId(int competenceId) { this.competenceId = competenceId; }

    public void setCategory(String category) { this.category = category; }
    public void setType(String type) { this.type = type; }
    public void setDescription(String description) { this.description = description; }

    public void setNiveau(String niveau) { this.niveau = niveau; }
    public void setAnneesExperience(int anneesExperience) { this.anneesExperience = anneesExperience; }
    public void setCertification(String certification) { this.certification = certification; }
    public void setStatut(String statut) { this.statut = statut; }

    public void setEmail(String email) { this.email = email; }
    public void setStatusChange(String statusChange) { this.statusChange = statusChange; }
    public void setDateModification(LocalDateTime dateModification) { this.dateModification = dateModification; }

    /* =========================
       CHECKBOX PROPERTY
       ========================= */

    public BooleanProperty selectedProperty() { return selected; }
    public boolean isSelected() { return selected.get(); }
    public void setSelected(boolean value) { selected.set(value); }
}

File: src/Entity/competence/HistoriqueCompetence.php
<?php

namespace App\Entity\competence;

use App\Repository\competence\HistoriqueCompetenceRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: HistoriqueCompetenceRepository::class)]
#[ORM\Table(name: 'historiques_competences')]
class HistoriqueCompetence
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(name: 'competence_id', type: 'integer', nullable: true)]
    private ?int $competenceId = null;

    #[ORM\Column(length: 100, nullable: true)]
    private ?string $category = null;

    #[ORM\Column(length: 100, nullable: true)]
    private ?string $type = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(length: 50, nullable: true)]
    private ?string $niveau = null;

    #[ORM\Column(name: 'annees_experience', type: 'integer', nullable: true)]
    private ?int $anneesExperience = 0;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $certification = null;

    #[ORM\Column(length: 50, nullable: true)]
    private ?string $statut = null;

    #[ORM\Column(length: 150, nullable: true)]
    private ?string $email = null;

    #[ORM\Column(name: 'status_change', length: 100, nullable: true)]
    private ?string $statusChange = null;

    #[ORM\Column(name: 'date_modification', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $dateModification = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCompetenceId(): ?int
    {
        return $this->competenceId;
    }

    public function setCompetenceId(?int $competenceId): self
    {
        $this->competenceId = $competenceId;
        return $this;
    }

    public function getCategory(): ?string
    {
        return $this->category;
    }

    public function setCategory(?string $category): self
    {
        $this->category = $category;
        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(?string $type): self
    {
        $this->type = $type;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getNiveau(): ?string
    {
        return $this->niveau;
    }

    public function setNiveau(?string $niveau): self
    {
        $this->niveau = $niveau;
        return $this;
    }

    public function getAnneesExperience(): ?int
    {
        return $this->anneesExperience;
    }

    public function setAnneesExperience(?int $anneesExperience): self
    {
        $this->anneesExperience = $anneesExperience;
        return $this;
    }

    public function getCertification(): ?string
    {
        return $this->certification;
    }

    public function setCertification(?string $certification): self
    {
        $this->certification = $certification;
        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(?string $statut): self
    {
        $this->statut = $statut;
        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(?string $email): self
    {
        $this->email = $email;
        return $this;
    }

    public function getStatusChange(): ?string
    {
        return $this->statusChange;
    }

    public function setStatusChange(?string $statusChange): self
    {
        $this->statusChange = $statusChange;
        return $this;
    }

    public function getDateModification(): ?\DateTimeInterface
    {
        return $this->dateModification;
    }

    public function setDateModification(?\DateTimeInterface $dateModification): self
    {
        $this->dateModification = $dateModification;
        return $this;
    }
}